File: Codeforces/1300/romanticglasses.py
t = int(input())

for _ in range(t):

    n = int(input())
    a = list(map(int, input().split()))

    curdif = 0
    diflist = []
    found = False

    for i, juice in enumerate(a):
        if i+1 & 1 == 1:
            curdif += juice
            if curdif not in diflist and curdif != 0:
                diflist.append(curdif)
            elif curdif in diflist or curdif == 0:
                found = True
                break
        else:
            curdif -= juice
            if curdif not in diflist and curdif != 0:
                diflist.append(curdif)
            elif curdif in diflist or curdif == 0:
                found = True
                break

    if found:
        print('yEs')
    else:
        print('nO')

File: Codeforces/1300/romanticglasses.cpp
#include <bits/stdc++.h> 

using namespace std;

int main(){

    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int t;
    cin >> t;

    while(t--){

        int n;
        cin >> n;

        vector<long long> a(n);
        for (int i = 0; i < n; i++){
            cin >> a[i];
        }

        long long curdif = 0;
        set<long long> difflist;
        bool found = false;

        for (int i = 0; i < n; i++){
            if (i+1 & 1 == 1){
                curdif += a[i];
                if (curdif != 0 && difflist.find(curdif) == difflist.end()) {
                difflist.insert(curdif);
                } 
                else {
                found = true;
                break;
                }
            }
            else {
                curdif -= a[i];
                if (curdif != 0 && difflist.find(curdif) == difflist.end()) {
                difflist.insert(curdif);
                } 
                else {
                found = true;
                break;
                }
            }

        }

        if (found){
            cout << "yEs" << endl;
        }
        else{
            cout << "nO" << endl;
        }


    }

    return 0;
}